oak-store-spi/src/main/java/org/apache/jackrabbit/oak/json/ParseUtils.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.json;
+
+import java.util.Arrays;
+
+public class ParseUtils {
+    /*
+     * Taken from https://github.com/google/guava/blob/v33.2.1/guava/src/com/google/common/primitives/Longs.java
+     */
+    private static final class AsciiDigits {
+        private AsciiDigits() {
+        }
+
+        private static final byte[] asciiDigits;
+
+        static {
```

Comment:
I believe this could be simplified even more, but maybe it doesn't matter a lot.

Reply:
I do not want to make too many changes to the code, otherwise we risk introducing bugs. Even a change that may seem safe, could introduce non-obvious problems. So unless there is a really good reason to change things, I'd rather leave it close to the original source code.